src/github.ts

```diff
@@ -166,23 +170,50 @@ export async function getRelease(
 
   /**
    * We need to:
-   *  - only consider releases starting with the provided `nameStartsWith`
-   *    and `isDraft` parameters
+   *  - only consider releases starting with the provided prefix, draft and
+   *    release parameters
+   *  - consider the major/minor constraint, if provided
    *  - _NOT_ rely on the temporal data; the precendence of the existing tags
    *    shall determined according to a "SemVer-esque prerelease", that is:
    *      * componentX-1.2.3-9 < componentX-1.2.3-10
    *  - return the highest-precedence item
    */
 
   const releaseList = result
-    .filter(r => r.isDraft === isDraft)
-    .filter(r => SemVer.fromString(r.tagName)?.prefix === prefixMustMatch)
+    .filter(r => r.isDraft === params.draftOnly)
+    .filter(r => {
+      const asSemVer = SemVer.fromString(r.tagName);
+      return asSemVer?.prefix === params.prefixToMatch &&
+        params.fullReleasesOnly
+        ? asSemVer?.prerelease === ""
+        : true;
+    })
     .map(r => ({ id: r.id, name: r.tagName }))
     .sort((lhs, rhs) => SemVer.sortSemVer(lhs.name, rhs.name));
 
   core.debug(
     `getRelease: sorted list of releases:\n${JSON.stringify(releaseList)}`
   );
+
+  if (params.constraint) {
+    // We're sorted by precedence, highest last, so let's reverse it and we can
+    // take the first major/minor version lower than the constraint we encounter.
+    releaseList.reverse();
```

Comment:
Should we `pop()` the last items in case `params.constraint` is set, but we do not have a matching semver based on the contstraints (iso the front items, which was previous logic)?

Reply:
This needs a `return` after the for-loop (or the `.pop()` should go into an else-branch). :+1: